src/main/java/io/fabric8/maven/docker/service/BuildService.java

```diff
@@ -388,6 +377,31 @@ private void autoPullBaseImage(ImageConfiguration imageConfig, ImagePullManager
         }
     }
 
+    private void autoPullCacheFromImage(ImageConfiguration imageConfig, ImagePullManager imagePullManager, BuildContext buildContext) throws DockerAccessException, MojoExecutionException {
+        if (imageConfig.getBuildConfiguration().getCacheFrom() == null) {
```

Comment:
`cacheFrom` is a List. In most cases the field is indeed `null` when the list is empty (not good practice IMHO), but don't you think you'd better to check for `getCacheFrom().isEmpty()` here as well (for robustness) ?

Same remark for `imageConfig.getBuildConfiguration()`. The method is invoked only when the buildConfig is not null but nothing in the method signature let you know that. Again, just for robustness, I would check if buildConfig!=null in the same if...

Reply:
- a check for `getCacheFrom().isEmpty()` is not needed, because it will never go inside the for-loop. The method will just return gracefully.
- The `imageConfig.getBuildConfiguration()` is never checked for nullness in this class already (5 occurences). I decided to skip this check to be consistent with the rest of the class.

Reply:
This is not strictly required indeed. Was just for code robustness in case of future refactoring... No problem for me with your approach.